File: ed/l/python/examples/blur/bench.py
import cv2


def b(f):
  img = cv2.imread(f)

  top, bottom, left, right = 286, 308, 111, 144
  rec = img[top:bottom, left:right]
  rec = cv2.GaussianBlur(rec, (3, 9), 5)
  img[top:bottom, left:right] = rec

  cv2.imwrite('img.after.py.png', img)


if __name__ == '__main__':
  b('./img.before.png')

File: ed/l/cpp/examples/blur/bench.cpp
/*
@example:
docker run -it --rm -v $PWD:/gh -w /gh spmallick/opencv-docker:opencv sh -c '
  f=ed/l/cpp/examples/blur/bench.cpp
  g++ -w $f -o x `pkg-config --cflags --libs opencv`
  ./x
'
*/

#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <stdio.h>
#include <iostream>

int b(std::string f)
{
    cv::Mat img = cv::imread(f, CV_LOAD_IMAGE_UNCHANGED);

    int top = 286, bottom = 308, left = 111, right = 144;
    cv::Rect r = cv::Rect(left, top, right - left, bottom - top);
    cv::Mat rec = img(r);
    cv::blur(rec, rec, cv::Size(3, 9));
    img(r) = rec;

    cv::imwrite(f+".r.png", img);
}

int main(int argc, char *argv[])
{
    b("/gh/ed/l/python/examples/blur/a.png");
    return 0;
}
